File: source/backend/project.Presentation/Models/Auth/AuthRequest.cs
namespace project.Presentation.Models.Auth
{
    public record ForgotPasswordRequest(string Email, string ClientUri);
    public record LoginRequest(string MSSV, string Password);
    public record ResetPasswordRequest(string Email, string Token, string NewPassword, string PasswordConfirm);
    public record ChangePasswordRequest(string OldPassword, string NewPassword, string PasswordConfirm);
    public record TokenRequest(string AccessToken, string RefreshToken);
    public record LinkGithubRequest(string email);

}

File: source/frontend/src/modules/auth/types/auth.request.ts
import type { UserRole } from './auth.enum'

export interface RegisterRequest {
  userName: string
  email: string
  password: string
  userCode?: string
  userRole?: UserRole
}

export interface LoginRequest {
  mssv: string
  password: string
}

export interface RefreshTokenRequest {
  refreshToken: string
}

export interface ForgotPasswordRequest {
  email: string
}

export interface ResetPasswordRequest {
  token: string
  newPassword: string
  confirmPassword: string
}

export interface ChangePasswordRequest {
  currentPassword: string
  newPassword: string
  confirmPassword: string
}

export interface UpdateProfileRequest {
  userName?: string
  avatarUrl?: string
}
